lib/firmata.js

```diff
@@ -209,20 +211,28 @@ SYSEX_RESPONSE[CAPABILITY_RESPONSE] = function(board) {
 
   // Only create pins if none have been previously created on the instance.
   if (!board.pins.length) {
+    var pin = {};
     for (var i = 2, n = 0; i < board.currentBuffer.length - 1; i++) {
       if (board.currentBuffer[i] === 127) {
-        board.pins.push({
-          supportedModes: supportedModes(capability),
-          mode: undefined,
-          value: 0,
-          report: 1,
-        });
+        pin.supportedModes = supportedModes(capability);
+        pin.mode = undefined;
+        pin.value = 0;
+        pin.report = 1;
+        board.pins.push(pin);
+        pin = {};
         capability = 0;
         n = 0;
         continue;
       }
       if (n === 0) {
         capability |= (1 << board.currentBuffer[i]);
+        pin[modeNames[board.currentBuffer[i]] + "Resolution"] =  board.currentBuffer[i + 1];
```

Comment:
Maybe a better strategy is to not apply these properties directly to the pin object here. Apply them to a temporary object and then apply to pin only in the above block. That way all assignment occurs within the same block much like how `capability` is applied to the pin in the above block as well.